Add sw-precache

denar90 opened the pull request:
Related to #469

denar90 commented:
This PR is ready for review. Before merging it we need to solve #454 (moving on https).
P.S. For reviewer. Please squash my commits 😄

denar90 commented:
Closing this until we can find solution for using HTTPS.

denar90 commented:
Reopening that due to #469. I'll resolve and retest stuff later...

paulfalgout commented:
For this we'll want to do something to redirect people to https:// right?  Like currently either http:// or https:// works.

denar90 commented:
@paulfalgout 👍

denar90 commented:
I've updated stuff. Also added some improvements. 

---

So right now we have 88% score for [Lighthouse](https://github.com/GoogleChrome/lighthouse/)
Using prod with `https` will higher this value up.

![image](https://cloud.githubusercontent.com/assets/6231516/24085050/721b8518-0cfd-11e7-9021-6e35250f7619.png)

---

To reach 100% we have to solve - https://github.com/marionettejs/marionettejs.com/issues/495

denar90 commented:
Since @thejameskyle added redirect to `https`, this PR ready for review :)

samccone commented:
💥 +1

denar90 commented on src/sections/_nav.jade:
```diff
@@ -1,4 +1,4 @@
-input.hidden-visually(type="checkbox", id="toggle-search")
+input.hidden-visually(type="checkbox", id="toggle-search", hidden) 
```
@peterblazejewicz yeap you are right 👍  I have to recheck if this stuff https://github.com/marionettejs/marionettejs.com/blob/master/src/js/search.js#L6 will work though.
